File: src/dataset/python/kindergarten-garden_exercism-fruit-in/openAI/ChatGPT4_kindergarten_garden_v2.py
class Garden:
    STUDENTS = ("Alice", "Bob", "Charlie", "David", "Eve", "Fred",
                "Ginny", "Harriet", "Ileana", "Joseph", "Kincaid", "Larry")
    PLANTS = {"G": "Grass", "C": "Clover", "R": "Radishes", "V": "Violets"}

    def __init__(self, diagram, students=STUDENTS):
        rows = diagram.splitlines()
        students = sorted(students)
        self.cups = {
            students[i]: [self.PLANTS[rows[row][2 * i + col]] 
                          for row in range(2) for col in range(2)]
            for i in range(len(rows[0]) // 2)
        }

    def plants(self, student):
        return self.cups.get(student, [])

File: src/dataset/javascript/kindergarten-garden_exercism-ThomasZumsteg/openAI/ChatGPT4_kindergarten-garden_v2.js
const DEFAULT_STUDENTS = ['Alice', 'Bob', 'Charlie', 'David', 'Eve', 
    'Fred', 'Ginny', 'Harriet', 'Ileana', 'Joseph', 'Kincaid', 'Larry'];

class Garden {
    constructor(garden, students = DEFAULT_STUDENTS) {
        this.students = [...students].sort().map(student => student.toLowerCase());
        this.gardenRows = garden.split('\n');
        this.plantMap = { 'C': 'clover', 'G': 'grass', 'R': 'radishes', 'V': 'violets' };

        this.students.forEach((student, index) => {
            this[student] = this.getStudentPlants(index);
        });
    }

    getStudentPlants(position) {
        const plants = [];
        for (const row of this.gardenRows) {
            plants.push(this.plantMap[row[2 * position]], this.plantMap[row[2 * position + 1]]);
        }
        return plants;
    }
}

module.exports = Garden;